index.js

```diff
@@ -1,53 +1,113 @@
-    return jasmine.getEnv().it.only(specName, runPitTest.bind(null, promiseBuilder));
-  };
+      var currentSpec = env.currentSpec;
 
-  globalObject.xpit = function xpit(specName, promiseBuilder) {
-    return jasmine.getEnv().xit(specName, runPitTest.bind(null, promiseBuilder));
-  };
+      currentSpec.runs(function () {
+        try {
+          var promise = promiseFn.call(currentSpec);
 
-  function runPitTest(promiseBuilder) {
-    var jasmineEnv = jasmine.getEnv();
-    var spec = jasmineEnv.currentSpec;
-    var isFinished = false;
-    var error = null;
+          assertPromise(promise);
 
-    spec.runs(function() {
-      try {
-        var promise = promiseBuilder.call(spec);
-        if (!promise || !promise.then) {
-          throw new Error('pit() tests should return a promise');
+          promise.then(function () {
+            isFinished = true;
+          })['catch'](function (err) {
+            error = err;
+            isFinished = true;
+          });
+        } catch (e) {
+          error = e;
+          isFinished = true;
         }
+      });
 
-        promise.then(function() {
-          isFinished = true;
-        })['catch'](function(err) {
-          error = err; isFinished = true;
-        });
-      } catch (e) {
-        error = e;
-        isFinished = true;
+      currentSpec.waitsFor(function () {
+        return isFinished;
+      });
+
+      currentSpec.runs(function () {
+        if (error) throw error;
+      });
+    };
+  }
+
+  function wrapPromiseJasmine(promiseFn) {
+    return function (done) {
+      var promise = promiseFn();
+      var fail = done.fail ? done.fail : function (err) { throw err; };
+
+      try {
+        assertPromise(promise);
+      } catch (err) {
+        fail(err);
       }
-    });
 
-    spec.waitsFor(function() { return isFinished; });
-    spec.runs(function() { if (error) throw error; });
-  };
-}
+      promise.then(function (res) {
+        done(res);
+      })['catch'](function (err) {
+        fail(err);
+      });
+    };
+  }
+
+  function install(globalObject) {
+    var jasmine = globalObject.jasmine;
 
-exports.install = install;
+    assert(
+      'It looks like you\'re trying to install jasmine-pit before installing ' +
+      'jasmine! Make sure there is a `jasmine` property on the global object ' +
+      '(window/global/etc) before calling install().', jasmine);
+
+    var env = jasmine.getEnv();
+    var isLegacyJasmine = jasmine.version_ && jasmine.version_.major === 1;
+    var wrapFn = isLegacyJasmine ? wrapPromiseLegacyJasmine.bind(null, env) : wrapPromiseJasmine;
+
+    globalObject.pit = function pit(specName, promiseFn) {
+      return env.it(specName, wrapFn(promiseFn));
+    };
+
+    globalObject.xpit = function xpit(specName, promiseFn) {
+      return env.xit(specName, wrapFn(promiseFn));
+    };
+
+    if (isLegacyJasmine) {
+      globalObject.pit.only = function pitOnly(specName, promiseFn) { // TODO: add note for jasmine-only
```

Comment:
Jasmine 2.x has `fit`and `fdescribe` that serves the same purpose.

Note left because `jasmine-pit` doesn't depends on `jasmine-only` but uses it. I haven't changed `dependencies` in `package.json`